src/web3py/extensions/tx_utils.py

```diff
@@ -79,7 +82,10 @@ def _estimate_gas(transaction: ContractFunction, account: LocalAccount) -> Optio
         try:
             gas = transaction.estimate_gas({'from': account.address})
         except ContractLogicError as error:
-            logger.warning({'msg': 'Contract logic error', 'error': str(error)})
+            logger.warning({'msg': 'Contract logic error.', 'error': str(error)})
+            return None
+        except ValueError as error:
+            logger.warning({'msg': 'Execution reverted.', 'error': str(error)})
```

Comment:
is this the only possible cause of the ValueError?

Reply:
We will see details in "error" and if it is another exception we will see it
But for now as I know yes